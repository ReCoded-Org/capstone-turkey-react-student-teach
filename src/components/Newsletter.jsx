import React from 'react';
import { useState } from 'react';

const Newsletter = () => {

  const [inputValue, setInputValue] =  useState("");

  const handleChange = (e) => {
    setInputValue(e.currentTarget.value)
    e.preventDefault();
  }

  return (
    <>
      <form className='block flex-col justify-items-center align-middle content-center md:grid-cols-1 lg:grid-cols-1 xl:grid-cols-1'>

        <input 
          type="text"
          value={inputValue}
          onChange={handleChange}
          placeholder="Your Email Address"
          className="block w-[312px] h-auto text-base font-semibold transition ease-in-out outline outline-1 outline-secondary-color rounded-md text-font-color px-2 py-2 mb-3 text-center justify-center border-secondary-color"
        />

        <a href="#" className="inline-flex items-center justify-center w-[468px]">
          <button
            type="submit"
            className="block w-[468px] h-auto text-base border border-transparent font-semibold text-white bg-primary-color hover:bg-white hover:text-primary-color hover:border-primary-color rounded-md px-5 py-3">
              Submit
          </button>
        </a>

      </form>
    </>
  )
}

export default Newsletter