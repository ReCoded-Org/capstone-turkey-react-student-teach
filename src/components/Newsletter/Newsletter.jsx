import * as Yup from 'yup';
import { Formik, Field, Form } from 'formik';
import { useSelector } from 'react-redux';

function Newsletter() {
  const darkMode = useSelector((state) => state.darkModeReducer.darkMode);

  const validateEmail = (value) => {
    let error;
    if (!value) {
      error = 'Required!';
    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value)) {
      error = 'Enter a proper email address...';
    }
    return error;
  };

  const validationSchema = Yup.object().shape({
    email: Yup.string().email('Invalid email address...').required('Required'),
  });

  const handleSubmit = () => {};

  return (
    <Formik
      initialValues={{
        email: '',
      }}
      validationSchema={validationSchema}
      onSubmit={handleSubmit}
      validator={() => ({})}
    >
      {({ errors, touched }) => (
        <Form className="flex flex-col justify-center items-center md:grid-cols-1 lg:grid-cols-1 xl:grid-cols-1 2xl:grid-cols-1 grid-cols-1">
          <Field
            id="email"
            name="email"
            type="email"
            validate={validateEmail}
            placeholder="Your Email Address"
            className=" lg:w-[312px] h-auto text-base font-semibold transition ease-in-out outline outline-1 outline-secondary-color rounded-md text-font-color px-2 py-2 mb-3  text-center border-secondary-color
            focus:invalid:border-primary-color focus:border-2 focus:border-primary-color xs:w-[270px] xs-justify-center xs:justify-center xs:text-center xs:align-middle mx-auto"
          />
          {errors.email && touched.email ? (
            <div className="text-center mb-4">{errors.email}</div>
          ) : null}
          <a href="!#">
            <button
              type="submit"
              className={`lg:w-[468px] h-auto text-base border border-transparent font-semibold text-white  hover:bg-white rounded-md px-5 py-3 w-[auto] xs:justify-center xs:align-middle ${
                darkMode ? 'hover:text-cusOrange' : 'hover:text-primary-color'
              } ${
                darkMode ? 'hover:border-white' : 'hover:border-primary-color'
              } ${darkMode ? 'bg-cusOrange' : 'bg-primary-color'} ${
                darkMode ? 'border-white' : 'bg-primary-color'
              }`}
            >
              Submit
            </button>
          </a>
        </Form>
      )}
    </Formik>
  );
}

export default Newsletter;
