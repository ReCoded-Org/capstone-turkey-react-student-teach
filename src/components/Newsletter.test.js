import React from 'react';
import renderer from 'react-test-renderer';
import Newsletter from './Newsletter';

test('Initial test', () => {
  const tree = renderer.create(<Newsletter />).toJSON();

  expect(tree).toMatchSnapshot();
});
