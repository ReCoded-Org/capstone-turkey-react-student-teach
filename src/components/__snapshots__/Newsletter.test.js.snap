// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial test 1`] = `
<form
  action="#"
  className="flex flex-col justify-self-center md:grid-cols-1 lg:grid-cols-1 xl:grid-cols-1 2xl:grid-cols-1 grid-cols-1"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <input
    className=" w-[312px] h-auto text-base font-semibold transition ease-in-out outline outline-1 outline-secondary-color rounded-md text-font-color px-2 py-2 mb-3 ml-[89px] text-center border-secondary-color"
    id="email"
    name="email"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Your Email Address"
    type="email"
    value=""
  />
  <a
    className="w-[468px]"
    href="#"
  >
    <button
      className="w-[468px] h-auto text-base border border-transparent font-semibold text-white bg-primary-color hover:bg-white hover:text-primary-color hover:border-primary-color rounded-md px-5 py-3"
      type="submit"
    >
      Submit
    </button>
  </a>
</form>
`;
